src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1058,6 +1060,47 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._terminalAltBufferActiveContextKey.set(!!(this.xterm && this.xterm.raw.buffer.active === this.xterm.raw.buffer.alternate));
 	}
 
+	private async _shouldPasteText(text: string): Promise<boolean> {
+		let textForLines = text.split(/(\r\n|\n|\r)/);
+		let confirmation: IConfirmationResult;
+
+		// If the clipboard has only one line, no prompt will be triggered
+		if (textForLines.length === 1 || this._configurationService.getValue<boolean>(TerminalSettingId.MultiLinePasteWarning) === false) {
+			confirmation = { confirmed: true };
+		} else {
+			let message = nls.localize('confirmMoveTrashMessageFilesAndDirectories', "Are you sure you want to paste the following {0} lines to the terminal?", text.split(/\r\n|\n|\r/).length);
+
+			let displayItemsCount = 3;
+			let detail = textForLines.slice(0, displayItemsCount).join('');
+			if (textForLines.length > displayItemsCount) {
+				detail += '...';
```

Comment:
Can use the ellipsis character here

```suggestion
				detail += '…';
```